lesson_10/eva-chitul/polygons_homework.py

```diff
@@ -0,0 +1,77 @@
+'''We have new customers for our Polygons company.
+    They need to create square objects with a certain area
+    They need a method to compute the perimeter only for triangle objects
+Requirements:
+    add an alternative constructor to Square class that takes the area as an argument and creates a square object with the apropriate sides
+    example :
+    >> sq = Square.from_area(8)
+    >> print(sq)
+    >> Side 1 with lenght: 2
+    >> Side 2 with lenght: 4
+    >> Side 3 with lenght: 2
+    >> Side 4 with lenght: 4
+    we want to make our perimeter method also available to other shapes in the future. Create a mixin class for perimeter that contains the perimeter method
+'''
+from math import sqrt
+import random
+
+
+class Polygon:
+    def __init__(self, *args):
+        self.sides = list(args)
+
+    def __str__(self):
+        return f'Polygon object has {len(self.sides)} sides'
+
+    def display(self):
+        for index, side in enumerate(self.sides, start=1):
+            print(f'Side {index} with length: {side}')
+
+
+class PerimeterMixin(Polygon):
+    def __init__(self, *args):
```

Comment:
un mixin nu are nevoie de `__init__` pentru ca nu o sa fie instantiat , mixinurile sunt doar mostenite. In cazul de fata nici mostenirea din Polygon nu este necesara, dar metoda care face perimetrul

Reply:
resolved and updated.